### PR TITLE
More flow

### PR DESCRIPTION
**Pull request by TomPridham**

### CODE REVIEW

File: src/each.js [hunk 1]
```diff
@@ -24,21 +23,26 @@ const iterateObject = (obj, fn) => {
...
 
-// $FlowFixMe
-const each = (list, fn) =>
-  isArrayLike(list)
-    ? iterateArray(list, fn)
-    : iterateObject(list, fn)
+const each = <T>(list: ?(*[] | {[string]: *}), fn: ?((?*, (number | string)) => T)) => {
+  if (list && fn) {
+    if (Array.isArray(list)
+    ) {
```
Comment (zacanger) [inline src/each.js]: This should be isArrayLike; right now `each` and `filter` also work for strings.

File: src/filter.js [hunk 2]
```diff
@@ -12,15 +11,16 @@ import each from './each'
  */
 
 const filter = (
-  fn: ((mixed) => bool, ?number) => bool,
+  fn: (*, (string | number)) => bool,
   list: mixed[] | Object
 ): Object | mixed[] => {
-  const isArr = isArrayLike(list)
+  const isArr = Array.isArray(list)
```
Comment (zacanger) [inline src/filter.js]: This should be isArrayLike; right now `each` and `filter` also work for strings.

File: src/until.js [hunk 3]
```diff
@@ -1,15 +0,0 @@
-// @flow
```
Comment (zacanger) [inline src/until.js]: This needs to stay for now. I can remove it in the next major version.

File: src/pipe.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ declare function pipe<A, B, C, D, T>((A) => B, (B) => C, (C) => D, (D) => T): T
 declare function pipe<A, B, C, D, E, T>((A) => B, (B) => C, (C) => D, (D) => E, (E) => T): T
 declare function pipe<A, B, C, D, E, F, T>((A) => B, (B) => C, (C) => D, (D) => E, (E) => F, (F) => T): T
 declare function pipe<A, B, C, D, E, F, G, T>((A) => B, (B) => C, (C) => D, (D) => E, (E) => F, (F) => G, (G) => T): T
-declare function pipe(Function): Function // eslint-disable-line flowtype/no-weak-types
+declare function pipe(Function): Function
```
Comment (zacanger) [inline src/pipe.js]: `Function` is a weak type, that comment should stay

File: src/promisify.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ declare function promisify<A, B, C, D, T>((A, B, C, D, Callback<T>) => void): (A
 declare function promisify<A, B, C, D, E, T>((A, B, C, D, E, Callback<T>) => void): (A, B, C, D, E) => Promise<T>
 declare function promisify<A, B, C, D, E, F, T>((A, B, C, D, E, F, Callback<T>) => void): (A, B, C, D, E, F) => Promise<T>
 declare function promisify<A, B, C, D, E, F, G, T>((A, B, C, D, E, F, G, Callback<T>) => void): (A, B, C, D, E, F, G) => Promise<T>
-declare function promisify(Function): Function // eslint-disable-line flowtype/no-weak-types
+declare function promisify(Function): Function
```
Comment (zacanger) [inline src/promisify.js]: Same here

File: src/uncurry.js [hunk 6]
```diff
@@ -8,7 +8,6 @@
  * add(1, 2) // => 3
  */
 
-// eslint-disable-next-line flowtype/no-weak-types
 const uncurry = (f: Function): Function =>
```
Comment (zacanger) [inline src/uncurry.js]: Same here, since this isn't really fixable

File: .flow-coverage-report.json [hunk 7]
```diff
@@ -8,6 +8,5 @@
   "includeGlob": [
     "src/**/*.js"
   ],
-  "threshold": 80,
```
Comment (zacanger) [inline .flow-coverage-report.json]: This should not be removed

File: .eslintrc.json [hunk 8]
```diff
@@ -4,7 +4,6 @@
   "parser": "babel-eslint",
   "rules": {
     "flowtype-errors/show-errors": [ 1 ],
-    "flowtype-errors/enforce-min-coverage": [ 1, 80 ],
```
Comment (zacanger) [inline .eslintrc.json]: This should not be removed

File: src/pipe.js (hunk 4 above)
Comment (TomPridham) [reply src/pipe.js]: it just disables a warning, so there isn't really a reason to leave it unless you don't want to update it to something more specific

**Pull request merged**